### PR TITLE
schema changed and controllers updated

### PR DESCRIPTION
**Pull request by Prybrt06**
Changed the datatype of the id from Int to string.

Since many to many relationships are not supported with mongoDB so created two new tables EventManager & EventOrganiser. In the EventManager Table user and the event that is managed by that user are stored. Similarly in EventOrganiser Table user and the event that is organized by that user are stored.
Screenshot 2023-12-28 at 12 13 09 AM
Screenshot 2023-12-28 at 12 13 12 AM
Screenshot 2023-12-28 at 12 13 29 AM

### CODE REVIEW

**Review by codadept — CHANGES_REQUESTED** [review]
Remove all the comments.

File: src/controllers/event/add.ts [hunk 1]
```diff
@@ -118,15 +148,41 @@ const createEvent: Interfaces.Controller.Async = async (req, res, next) => {
...
-      },
-      managers: {
-        connect: managersUsernames,
-      },
     },
   });
 
+  if (organizers) {
+    try {
+      organizers?.map(async (id: any) => {
```
Comment (JOSU9435) [inline src/controllers/event/add.ts]: remove the `?` operator before map as you are already checking for the presence of organizers 

If possible give `id` a more specific type than `any`, and try to not use `any` as much as possible

File: src/controllers/event/add.ts [hunk 2]
```diff
@@ -118,15 +148,41 @@ const createEvent: Interfaces.Controller.Async = async (req, res, next) => {
...
+      });
+    } catch (e) {
+      await prisma.event.delete({ where: { id: event.id } });
+      return next(Errors.Event.unableToCreate);
+    }
+  }
+
+  if (managers) {
+    try {
+      managers?.map(async (id: any) => {
```
Comment (JOSU9435) [inline src/controllers/event/add.ts]: Same changes as mentioned before for organizers

File: src/controllers/event/add.ts [hunk 3]
```diff
@@ -118,15 +148,41 @@ const createEvent: Interfaces.Controller.Async = async (req, res, next) => {
...
   });
 
+  if (organizers) {
+    try {
+      organizers?.map(async (id: any) => {
+        await prisma.eventOrganiser.create({
+          data: {
+            userId: id,
+            eventId: event.id,
+          },
```
Comment (JOSU9435) [inline src/controllers/event/add.ts:161]: Just to make sure, is this the correct way to create and relate entities in prisma ?

can you confirm ?

File: src/controllers/module/delete.ts [hunk 4]
```diff
@@ -9,8 +9,8 @@ const deleteModuleById: Interfaces.Controller.Async = async (
   next
 ) => {
   const { moduleId: MID } = req.params;
-  const moduleId = Number.parseInt(MID);
-  if (isNaN(moduleId)) return next(Errors.Module.invalidInput);
+  const moduleId = String(MID);
+  if (moduleId) return next(Errors.Module.invalidInput);
```
Comment (JOSU9435) [inline src/controllers/module/delete.ts]: wrong error condition, `!` is missing

**Review by JOSU9435 — CHANGES_REQUESTED** [review]

File: src/controllers/event/add.ts (hunk 1 above)
Comment (Prybrt06) [reply src/controllers/event/add.ts]: I have removed the `?` operator and gave the type `string` to the `id`.

File: src/controllers/event/add.ts (hunk 2 above)
Comment (Prybrt06) [reply src/controllers/event/add.ts]: Removed the `?` operator and gave the type `string` to the `id`.

File: src/controllers/event/add.ts (hunk 3 above)
Comment (Prybrt06) [reply src/controllers/event/add.ts:161]: Yes bhaiya it is working

File: src/controllers/module/delete.ts (hunk 4 above)
Comment (Prybrt06) [reply src/controllers/module/delete.ts]: Added `!`

File: src/controllers/event/get.ts [hunk 5]
```diff
@@ -106,7 +107,7 @@ const getEventById: Interfaces.Controller.Async = async (req, res, next) => {
   const { eventId: EID } = req.params;
   const eventId = String(EID);
 
-  if (!eventId) return next(Errors.Module.invalidInput);
+  if (!eventId || eventId.length != 24) return next(Errors.Module.invalidInput);
```
Comment (JOSU9435) [inline src/controllers/event/get.ts]: use `!==` instead of `!=`

File: src/controllers/event/add.ts [hunk 6]
```diff
@@ -46,6 +46,8 @@ const createEvent: Interfaces.Controller.Async = async (req, res, next) => {
   )
     return next(Errors.Module.invalidInput);
 
+  if (moduleId.length != 24) return next(Errors.Module.invalidInput);
```
Comment (JOSU9435) [inline src/controllers/event/add.ts]: use `!==` instead of `!=`

**Review by JOSU9435 — CHANGES_REQUESTED** [review]

File: src/controllers/event/get.ts (hunk 5 above)
Comment (Prybrt06) [reply src/controllers/event/get.ts]: done

File: src/controllers/event/add.ts (hunk 6 above)
Comment (Prybrt06) [reply src/controllers/event/add.ts]: done

**Review by JOSU9435 — APPROVED** [review]

**Pull request merged**